Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -590,7 +509,15 @@ public class HTTPClient {
                 connection.release(closing: true, logger: logger)
                 return channel.eventLoop.makeFailedFuture(error)
             }
-        }.always { _ in
+        }
+        .and(task.futureResult)
+        .always { result in
+            if case let .success((_, response)) = result, let httpResponse = response as? HTTPClient.Response {
+                span.attributes.http.statusCode = Int(httpResponse.status.code)
+                span.attributes.http.statusText = httpResponse.status.reasonPhrase
+                span.attributes.http.responseContentLength = httpResponse.body?.readableBytes ?? 0
+            }
```

Comment:
It should record error if it was failure

Reply:
Updated 👍